docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,297 @@
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+The `TRANSFORM` clause is used to specifies a Hive-style transform (SELECT TRANSFORM/MAP/REDUCE)
+query specification to transform the input by forking and running the specified script. Users can
+plug in their own custom mappers and reducers in the data stream by using features natively supported
+in the Spark/Hive language. e.g. in order to run a custom mapper script `map_script` and a custom
+reducer script `reduce_script` the user can issue the following command which uses the TRANSFORM
+clause to embed the mapper and the reducer scripts.
+
+Currently, Spark's script transform support two mode:
+
+    1. Without Hive: It means spark run Spark SQL without Hive support, in this mode, spark can use default 
+       `ROW FORMAT DELIMITED` by treating data as STRING.
+    2. WIth Hive: It means spark run Spark SQL with Hive support, in this mode, when spark use default format, 
+       it will be treated as Hive default fomat. And spark can use Hive supported SerDe to process data.
+
+In both modes with default format, columns will be transformed to STRING and delimited by tabs before feeding
+to the user script, Similarly, all `NULL` values will be converted to the literal string `\N` in order to
+differentiate `NULL` values from empty strings. The standard output of the user script will be treated as
+TAB-separated STRING columns, any cell containing only `\N` will be re-interpreted as a `NULL`, and then the
+resulting STRING column will be cast to the data type specified in the table declaration in the usual way.
+User scripts can output debug information to standard error which will be shown on the task detail page on hadoop.
+These defaults can be overridden with `ROW FORMAT DELIMITED`.
+
+### Syntax
+
+```sql
+rowFormat
+    : ROW FORMAT SERDE serde_class [ WITH SERDEPROPERTIES serde_props ]
+    | ROW FORMAT DELIMITED
+       [ FIELDS TERMINATED BY fields_terminated_char [ ESCAPED BY escapedBy ] ]
+       [ COLLECTION ITEMS TERMINATED BY collectionItemsTerminatedBy ]
+       [ MAP KEYS TERMINATED BY keysTerminatedBy ]
+       [ LINES TERMINATED BY linesSeparatedBy ]
+       [ NULL DEFINED AS nullDefinedAs ]  
+
+inRowFormat=rowFormat
+outRowFormat=rowFormat
+namedExpressionSeq = named_expression [ , ... ]
+
+transformClause:
+  SELECT [ TRANSFORM ( namedExpressionSeq ) | MAP namedExpressionSeq | REDUCE namedExpressionSeq ]
```

Comment:
Could you move this into the beginning of this syntax section?

Reply:
Done